apps/agent/app/src/components/useDownloadData.js

```diff
@@ -1,110 +1,100 @@
+import { formatDate, ISO_SHORT_FORMAT } from '../lib/date-utils'
+import { TRANSACTION_TYPES_LABELS } from '../transaction-types'
 
-const formatDate = date => format(date, 'MM/dd/yy')
+// Transforms a two dimensional array into a CSV data structure
+// Surround every field with double quotes + escape double quotes inside.
+function csv(data) {
+  return data
+    .map(cells => cells.map(cell => `"${cell.replace(/"/g, '""')}"`).join(','))
+    .join('\n')
+}
 
-const getDownloadData = async ({
-  transfers,
-  tokenDetails,
-  resolveAddress,
-  tokens,
-}) => {
-  const mappedData = await transfers.reduce(
+async function getDownloadData({ transactions, tokenDetails, resolveAddress }) {
+  const processedTransactions = await transactions.reduce(
     async (
-      promise,
-      { tokenTransfers, date, description, type, isIncoming, onlyOne }
+      transactionListPromise,
+      { date, description, tokenTransfers, type }
     ) => {
-      const previous = await promise
+      const previous = await transactionListPromise
       const mappedTokenTransfersData = await Promise.all(
         tokenTransfers.map(async ({ amount, from, to, token }) => {
           const { symbol, decimals } = tokenDetails[toChecksumAddress(token)]
           const formattedAmount = formatTokenAmount(
             amount,
-            !!from,
+            Boolean(from),
             decimals,
             true,
-            { rounding: 5 }
+            { rounding: ROUNDING_AMOUNT }
+          )
+          const [source, recipient] = await Promise.all(
+            [from, to].map(address => {
+              return address
+                ? resolveAddress(address).then(
+                    res => (res && res.name) || address
+                  )
+                : 'Agent'
+            })
           )
-          const name = await (async () => {
-            if (!from) {
-              return 'Agent'
-            }
-            const { name } = (await resolveAddress(from)) || { name: '' }
-            return name
-          })()
-          const entity = await (async () => {
-            if (!to) {
-              return 'Agent'
-            }
-            const { name } = (await resolveAddress(to)) || { name: '' }
-            return name
-          })()
-          return `${formatDate(
-            date
-          )},${name},${entity},${description},${`"${formattedAmount} ${symbol}"`}`
+          return [
+            formatDate(date),
+            source,
+            recipient,
+            TRANSACTION_TYPES_LABELS[type],
+            description,
+            `${formattedAmount} ${symbol}`,
+          ]
         })
       )
       return [...previous, ...mappedTokenTransfersData]
     },
     /* https://gyandeeps.com/array-reduce-async-await/ */
     Promise.resolve([])
   )
-  return ['Date,Name,Source/Recipient,Reference,Amount']
-    .concat(mappedData)
-    .join('\n')
+
+  return csv([
+    ['Date', 'Source', 'Recipient', 'Type', 'Reference', 'Amount'],
```

Comment:
Let's add the transaction hash here (should be available from each transaction object)